source/multigrid/mg_transfer_internal.cc

```diff
@@ -197,24 +203,60 @@ namespace internal
             tria->level_ghost_owners();
           std::map<int, std::vector<DoFPair>> send_data;
 
-          // * find owners of the level dofs and insert into send_data
-          // accordingly
-          for (const auto &dofpair : send_data_temp)
+          sort(send_data_temp.begin(),
+               send_data_temp.end(),
+               [](const DoFPair &lhs, const DoFPair &rhs) {
+                 if (lhs.level < rhs.level)
+                   return true;
+                 if (lhs.level > rhs.level)
+                   return false;
+
+                 if (lhs.level_dof_index < rhs.level_dof_index)
+                   return true;
+                 if (lhs.level_dof_index > rhs.level_dof_index)
+                   return false;
+
+                 if (lhs.global_dof_index < rhs.global_dof_index)
+                   return true;
+                 else
+                   return false;
+               });
+          send_data_temp.erase(std::unique(send_data_temp.begin(),
+                                           send_data_temp.end()),
```

Comment:
For consistency with the lambda you use for `std::sort`, please also use a lambda here rather than the separate (old-style) `operator==` implementation in lines 56-62.